sdk/src/main/java/com/kin/ecosystem/gifting/GiftingManagerImpl.kt

```diff
@@ -0,0 +1,96 @@
+package com.kin.ecosystem.gifting
+
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import android.os.Message
+import com.kin.ecosystem.JwtProvider
+import com.kin.ecosystem.common.KinTheme
+import com.kin.ecosystem.common.ObservableData
+import com.kin.ecosystem.common.Observer
+import com.kin.ecosystem.common.exception.KinEcosystemException
+import com.kin.ecosystem.common.model.OrderConfirmation
+import com.kin.ecosystem.core.bi.EventLogger
+import com.kin.ecosystem.core.bi.events.APageViewed
+import com.kin.ecosystem.core.bi.events.GiftingButtonTapped
+import com.kin.ecosystem.core.bi.events.GiftingFlowCompleted
+import com.kin.ecosystem.core.bi.events.PageCloseTapped
+import com.kin.ecosystem.core.data.blockchain.BlockchainSource
+import com.kin.ecosystem.core.data.internal.Configuration
+import com.kin.ecosystem.core.data.order.OrderDataSource
+import org.kin.ecosystem.appreciation.options.menu.ui.CloseType
+import org.kin.ecosystem.appreciation.options.menu.ui.DialogTheme
+import org.kin.ecosystem.appreciation.options.menu.ui.EventsListener
+import org.kin.ecosystem.appreciation.options.menu.ui.GiftingDialog
+
+internal class GiftingManagerImpl(private val jwtProvider: JwtProvider,
+                                  private val blockchainSource: BlockchainSource,
+                                  private val orderDataSource: OrderDataSource,
+                                  private val eventLogger: EventLogger,
+                                  private val configuration: Configuration) : GiftingManager, EventsListener {
+
+	private val orderConfirmation: ObservableData<OrderConfirmation> = ObservableData.create()
+	private var currentRecipientUserID: String? = null
+	private var isShowing = false
+	private val mainHandler = object : Handler(Looper.getMainLooper()) {
```

Comment:
Just checking if you are sure you want to do it in the UI thread

Reply:
yup posting a callback on the main thread